internal/api/auth/auth.go

```diff
@@ -0,0 +1,189 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/brocaar/lorawan"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// Claims defines the struct containing the token claims.
+type Claims struct {
```

Comment:
Did you consider using [`StandardClaims`][1]? I see why you might want to implement your own solution, but I think that embedding this structure could also provide some benefits.

[1]: https://github.com/dgrijalva/jwt-go/blob/d2709f9f1f31ebcda9651b03077758c1f3a0018c/claims.go#L18